make/DynamoRIOConfig.cmake.in

```diff
@@ -1056,7 +1057,7 @@ function (configure_DynamoRIO_standalone target)
   # flags removed.  Usually the target LINK_FLAGS is empty at this point anyway.
   set(extra_flags "-DDYNAMORIO_STANDALONE")
   if (ANDROID)
-    set(extra_flags ${extra_flags} "-fPIE -pie")
```

Comment:
this change may be unnecessary.

Reply:
With `-pie`, the compiler complains that the argument is unused:
`clang: error: argument unused during compilation: '-pie' [-Werror,-Wunused-command-line-argument]`

GCC says 

> Options starting with -g, -f, -m, -O, -W, or --param are automatically
 passed on to the various sub-processes invoked by gcc.

So I imagine `-fPIE` just passes the `-PIE` option to the correct sub-process (I think the linker in this case), which is why clang complains about `-pie`, as it is not used for actual compilation. 

As for `-pie` vs `-PIE`, there is a [discussion](https://stackoverflow.com/questions/3544035/what-is-the-difference-between-fpic-and-fpic-gcc-parameters) on `-fpic` vs `-fPIC` which I imagine would mirror `pie/PIE`.

Reply:
@derekbruening @jimmypury does this resolve this comment?

Reply:
In my old notes there are multiple references to `-fPIE -pie` or `-fpie -pie` in build commands: including a pointer to here https://stackoverflow.com/questions/24818902/running-a-native-library-on-android-l-error-only-position-independent-executab.  I assume this change would not show up in a build failure and so we can't tell whether it has a negative effect on 32-bit Android as we only have build testing of that.  Are resulting binaries from this change clearly PIE binaries?  If so then may as well go with it; maybe add a comment.

Reply:
So I've done a bit more digging - there is another place in the code (in the main `CMakeLists.txt`) where `-fPIE and -pie` are enabled (for Android), and the flags get added to the linker.

The code executed here in `DynamoRIOConfig.cmake.in` adds the flags to a compilation command that includes `-c`, i.e. doesn't do any linking.  Since `-pie` is a linker option, Clang complains that it's unused (I guess GCC ignores it).

I think it's then correct to remove `-pie` here, as the process becomes:
1. Compile (without linking) with `-fPIE` only
2. Link with `-fPIE` and `-pie`

This seems to mirror the instructions in the [link](https://stackoverflow.com/questions/24818902/running-a-native-library-on-android-l-error-only-position-independent-executab) you provided: 
> LOCAL_CFLAGS += -fPIE
LOCAL_LDFLAGS += -fPIE -pie

I've build arm32 Android with this change and looking at the build output the only difference is the removal of `-pie` from 3 of the aforementioned `-c` compilation commands, and they all have `-pie` in the subsequent commands that link them. So things that need to be `-pie`'d are still being `-pie`'d, but just when linked as intended. So should be good to go, although I can still add a comment if necessary.

Reply:
But the pieces of CMake code you are referring to are probably not actually connected: the main CMakeLists.txt is used when building DR itself while make/DynamoRIOConfig.cmake.in is exported to users for their use when building a binary that links against DR.  The external user using this config code will not be pulling in any of DR's own internal CMake code (unless it's in the exported functions used by this config file).

Reply:
I see. I've changed it so that 32-bit Android keeps both `-fPIE` and `-pie` here, and 64-bit only has `-fPIE` with a comment about how Clang complains about `-pie`. 

`file` seems to think the Android64 build is still `pie`'d as below, but there are probably still compiler flag issues about for Android64.

> bin64/drrun: ELF 64-bit LSB pie executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /system/bin/linker64, not stripped

> lib64/release/libdynamorio.so: ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), static-pie linked, not stripped

Reply:
@derekbruening is this good to go now?

Reply:
Works for me.  (Didn't realize you were waiting: clicking on "re-review" the circular arrow is the best way to indicate you want more feedback.)